test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/architecture/EmptyUnmanagedKieServersWithSmartRouterIntegrationTest.java

```diff
@@ -0,0 +1,242 @@
+    }
+
+    @Test
+    @Ignore("https://issues.jboss.org/browse/RHBPMS-4899 && https://issues.jboss.org/browse/JBPM-6623")
+    public void testWorkingWithSignalsAndTasksDeployments() {
+        logger.debug("Deploy and start containers on Kie servers");
+        deployAndStartContainer(kieServerClientABC, CONTAINER_ID_ABC, deploymentScenario.getKieServerDeployments().get(0));
+        deployAndStartContainer(kieServerClientDEF, CONTAINER_ID_DEF, deploymentScenario.getKieServerDeployments().get(1));
+        deployAndStartContainer(kieServerClientGHI, CONTAINER_ID_GHI, deploymentScenario.getKieServerDeployments().get(2));
+
+        waitUntilKieServerLogsContain(deploymentScenario.getKieServerDeployments().get(0), "Container cont-id-abc (for release id org.kie.server.testing:definition-project-snapshot:1.0.0-SNAPSHOT) successfully started");
+        waitUntilKieServerLogsContain(deploymentScenario.getKieServerDeployments().get(1), "Container cont-id-def (for release id org.kie.server.testing:definition-project-snapshot:1.0.0-SNAPSHOT) successfully started");
+        waitUntilKieServerLogsContain(deploymentScenario.getKieServerDeployments().get(2), "Container cont-id-ghi (for release id org.kie.server.testing:definition-project-snapshot:1.0.0-SNAPSHOT) successfully started");
+
+        logger.debug("Start process instance");
+        List<Long> signalProcessInstances = createSignalProcesses();
+
+        assertThat(smartQueryServicesClient.findProcessInstances(0, 10)).hasSize(6);
+
+        smartProcessServicesClient.signalProcessInstances(CONTAINER_ID_ABC, signalProcessInstances, SIGNAL_NAME, null);
+        smartProcessServicesClient.signalProcessInstance(CONTAINER_ID_DEF, signalProcessInstances.get(2), SIGNAL_NAME, null);
+        smartProcessServicesClient.signalProcessInstance(CONTAINER_ID_DEF, signalProcessInstances.get(3), SIGNAL_2_NAME, null);
+
+        assertThat(smartTaskServicesClient.findTasks(USER_YODA, 0, 10)).hasSize(3);
+
+        smartProcessServicesClient.signal(CONTAINER_ID_ABC, SIGNAL_2_NAME, null);
+        assertThat(smartQueryServicesClient.findProcessInstancesByStatus(Arrays.asList(org.kie.api.runtime.process.ProcessInstance.STATE_COMPLETED), 0, 10)).hasSize(2);
+
+        assertThat(smartQueryServicesClient.findProcessInstancesByStatus(Arrays.asList(org.kie.api.runtime.process.ProcessInstance.STATE_ACTIVE), 0, 10)).hasSize(4);
+        List<TaskSummary> activeTasks = smartTaskServicesClient.findTasks(USER_YODA, 0, 10);
+        assertThat(activeTasks).hasSize(1);
+        assertThat(activeTasks.get(0).getContainerId()).isEqualTo(CONTAINER_ID_DEF);
+        assertThat(activeTasks.get(0).getProcessInstanceId()).isEqualTo(signalProcessInstances.get(2));
+
+        smartTaskServicesClient.startTask(CONTAINER_ID_DEF, activeTasks.get(0).getId(), USER_YODA);
+        smartTaskServicesClient.completeTask(CONTAINER_ID_DEF, activeTasks.get(0).getId(), USER_YODA, Collections.EMPTY_MAP);
```

Comment:
It is better to use Collections.emptyMap() instead of Collections.EMPTY_MAP for typed parameters  - to get rid of type safety warning.